packages/apollo-federation/src/composition/validate/sdl/__tests__/matchingEnums.test.ts

```diff
@@ -0,0 +1,206 @@
+            BED
+            BATH
+          }
+        `,
+        name: 'serviceA',
+      },
+      {
+        typeDefs: gql`
+          enum ProductCategory {
+            BEYOND
+          }
+        `,
+        name: 'serviceB',
+      },
+    ];
+
+    const definitionsDocument = createDefinitionsDocumentForServices(
+      serviceList,
+    );
+    const errors = validateSDL(definitionsDocument, schema, [matchingEnums]);
+    expect(errors).toMatchInlineSnapshot(`
+      Array [
+        [GraphQLError: The \`ProductCategory\` enum does not have identical values in all services. Groups of services with identical values are: [serviceA], [serviceB]],
+      ]
+    `);
+  });
+
+  it('errors when enums in separate services dont match', () => {
+    const serviceList = [
+      {
+        typeDefs: gql`
+          type Query {
+            products: [Product]!
+          }
+
+          type Product @key(fields: "sku") {
+            sku: String!
+            upc: String!
+            type: ProductType
+          }
+
+          enum ProductType {
+            BOOK
+            FURNITURE
+          }
+        `,
+        name: 'serviceA',
+      },
+      {
+        typeDefs: gql`
+          enum ProductType {
+            FURNITURE
+            BOOK
+            DIGITAL
+          }
+        `,
+        name: 'serviceB',
+      },
+      {
+        typeDefs: gql`
+          enum ProductType {
+            FURNITURE
+            BOOK
+            DIGITAL
+          }
+        `,
+        name: 'serviceC',
+      },
+    ];
+
+    const definitionsDocument = createDefinitionsDocumentForServices(
+      serviceList,
+    );
+    const errors = validateSDL(definitionsDocument, schema, [matchingEnums]);
+    expect(errors).toMatchInlineSnapshot(`
+      Array [
+        [GraphQLError: The \`ProductType\` enum does not have identical values in all services. Groups of services with identical values are: [serviceA], [serviceB, serviceC]],
+      ]
+    `);
+  });
+
+  it('errors when an enum name is defined as another type in a service', () => {
+    const serviceList = [
+      {
+        typeDefs: gql`
+          enum ProductType {
+            BOOK
+            FURNITURE
+          }
+        `,
+        name: 'serviceA',
+      },
+      {
+        typeDefs: gql`
+          type ProductType {
+            id: String
+          }
+        `,
+        name: 'serviceB',
+      },
+      {
+        typeDefs: gql`
+          enum ProductType {
+            FURNITURE
+            BOOK
+            DIGITAL
+          }
+        `,
+        name: 'serviceC',
+      },
+    ];
+
+    const definitionsDocument = createDefinitionsDocumentForServices(
+      serviceList,
+    );
+    const errors = validateSDL(definitionsDocument, schema, [matchingEnums]);
+    expect(errors).toMatchInlineSnapshot(`
+      Array [
+        [GraphQLError: [serviceA] ProductType -> ProductType is an enum in [serviceA, serviceC], but not in [serviceB]],
```

Comment:
This error could be clearer (the functionality is great)! Can we tell what kind of type it is in `serviceB` in this case? I.e. `ProductType is an enum in [serviceA, serviceC], but is an object type in [serviceB]`

Reply:
I'm not sure. What if there were multiple services with `ProductType` and one was an object type, and the other was a Scalar?